numpy/ma/core.pyi

```diff
@@ -1154,7 +1154,10 @@ def sort(
     *,
     stable: Literal[False] | None = False,
 ) -> NDArray[Any]: ...
-def compressed(x): ...
+@overload
+def compressed(x: _ArrayLike[_ScalarT_co]) -> ndarray[tuple[int], dtype[_ScalarT_co]]: ...
+@overload
+def compressed(x: ArrayLike) -> ndarray[tuple[int], Any]: ...
```

Comment:
there are some edge-cases related to overloads and protocols where this can matter

```suggestion
def compressed(x: ArrayLike) -> ndarray[tuple[int], dtype[Any]]: ...
```